### PR TITLE
Add tests for theme_extension.1.dart API example.

### PR DESCRIPTION
**Pull request by ksokolovskyi**
This PR contributes to https://github.com/flutter/flutter/issues/130459

### Description
- Adds tests for `examples/api/lib/material/theme/theme_extension.1.dart`

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [X] All existing and new tests are passing.

### CODE REVIEW

**Comment by ksokolovskyi** [comment]
@Renzo-Olivares, could you please take a look at this PR?

File: examples/api/test/material/theme/theme_extension.1_test.dart [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+    ThemeData theme = Theme.of(
+      tester.element(find.byType(example.Home)),
+    );
+    example.MyColors colors = theme.extension<example.MyColors>()!;
+
+    expect(colors.brandColor, equals(const Color(0xFF1E88E5)));
+    expect(colors.danger, equals(const Color(0xFFE53935)));
+
+    await tester.tap(find.byType(IconButton));
```
Comment (Renzo-Olivares) [inline examples/api/test/material/theme/theme_extension.1_test.dart:38]: nit: add a comment indicating what the tap is doing, in this case I think switching the theme to dark mode.

**Review by Renzo-Olivares — APPROVED** [review]
LGTM, thanks for adding this test!

**Comment by ksokolovskyi** [comment]
@Renzo-Olivares thanks for the review!

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Comment by ksokolovskyi** [comment]
@Renzo-Olivares @gspencergoog could you please tell me why Google testing is failing?

**Comment by ksokolovskyi** [comment]
@Renzo-Olivares still failing =(

**Comment by Renzo-Olivares** [comment]
@ksokolovskyi can you try, assuming you have `upstream` as one of your remote repos.

```
git fetch upstream
git rebase upstream/master
git push -f origin myBranch
```

**Comment by ksokolovskyi** [comment]
@Renzo-Olivares @gspencergoog @goderbauer thanks for the review!

**Pull request merged**